In [1]:
import pandas as pd
import plotly.express as px

In [2]:
df = pd.read_csv("Compute_allocations_transformer.csv")
# byte_level = df[df['Patch size'] == 1]
byte_level = df[(df['FLOPs'] == 's') & (df['Patch size'] == 2)]
# byte_level['Global transformer parameters'].sort_values(ascending=False)
fig = px.histogram(byte_level['Global model parameters'])

fig.show()

In [6]:
byte_level[(byte_level['Global model parameters'] <= 6.2e7) & (byte_level['Global model parameters'] >= 5.8e7)]

,FLOPs,Patch size,Tokens,Encoder parameters,Global model parameters,Decoder parameters,Global model dimension,Encoder/Decoder dimension,Global model layers,Decoder layers
202,s,2,1.297946e+10,2410455.04,58195271.68,4820910.08,832,448,7,2
206,s,2,1.225186e+10,3148349.44,58195271.68,6296698.88,832,512,7,2
372,s,2,1.222831e+10,2410455.04,59523481.60,4820910.08,704,448,10,2
376,s,2,1.158039e+10,3148349.44,59523481.60,6296698.88,704,512,10,2
472,s,2,1.300425e+10,1770946.56,58195271.68,5312839.68,832,384,7,3
609,s,2,1.225031e+10,1770946.56,59523481.60,5312839.68,704,384,10,3
685,s,2,1.201516e+10,1770946.56,59031552.00,5312839.68,640,384,12,3
788,s,2,1.152984e+10,1770946.56,59769446.40,5312839.68,576,384,15,3
863,s,2,1.252964e+10,1229824.00,59523481.60,4919296.00,704,320,10,4
867,s,2,1.170016e+10,1770946.56,59523481.60,7083786.24,704,384,10,4


4e18 FLOPs
===

tokens, global model params, decoder params, global model dimension, encoder/decoder dimension, global model layers, decoder layers

Patch size = 2

4.094315e+10, 11247452,	1224212, 448, 256, 6, 2

1.993053e+10 31483494, 3689472, 512, 320, 10, 3 (30M run)

1.188846e+10, 59031552, 6149120, 640, 320, 12, 5 (60M run)

Patch size = 4

4.937337e+10, 21691515, 1836319, 576, 256, 7, 3

1.847138e+10, 60177695, 5508956, 704, 384, 13, 4

9.375604e+09, 120M, 12242125, 896, 512, 16, 5